src/entity/posts/utils.ts

```diff
@@ -36,6 +39,75 @@ type StringPostStats = {
   [Property in keyof PostStats]: string;
 };
 
+interface SavePostProps<T extends Post> {
+  con: DataSource | EntityManager;
+  type: EntityTarget<T>;
+  post: DeepPartial<T>;
+  shouldUpdateFlags?: boolean;
+}
+
+export const savePost = async <T extends Post>({
+  con,
+  type,
+  post,
+  shouldUpdateFlags,
+}: SavePostProps<T>) => {
+  const res = await con.getRepository(type).save(post);
+
+  if (shouldUpdateFlags) {
+    const source = await con
+      .getRepository(Source)
+      .findOneBy({ id: res.sourceId });
+
+    const currentTotalPosts = source.flags?.totalPosts || 0;
+    const flags = updateFlagsStatement({
+      totalPosts: currentTotalPosts + 1,
+    });
```

Comment:
Might be out of loop but did we not say we are going to do this with triggers? This way counts will be updated no matter from where the post is deleted (API, script, psql etc.) we do similar for `upvotes` and `downvotes` for posts.

FYI @DragosIorgulescu

Reply:
Yeah, DB triggers are a good option as well. Looking forward to the reply as the ticket mentioned the mutation 👀

Reply:
I just saw the note about DB triggers being an option. I can maybe try and see how we can incorporate it. However, it only mentions the `totalPosts` to be part of it and the others would be within a cron job.

Reply:
The RFC highlights updating the mutations, we can maybe reopen a ticket if we ever decide to finalize using DB triggers. @capJavert feel free to make the final call, I am cool either way and willing to make the changes if necessary.